handler.go

```diff
@@ -53,21 +53,29 @@ func (a *podInjector) Handle(ctx context.Context, req admission.Request) admissi
 	handlerLog.V(9).Info("request=" + string(marshaled))
 
 	if req.Operation == admv1.Delete {
-		secret := &corev1.Secret{
-			TypeMeta: metav1.TypeMeta{
-				Kind:       "Secret",
-				APIVersion: "v1",
-			},
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      fmt.Sprintf("%s-%s", telegrafSecretPrefix, req.Name),
-				Namespace: req.Namespace,
-			},
+		deleteFailed := false
+		for _, name := range a.SidecarHandler.telegrafSecretNames(req.Name) {
```

Comment:
what is `req.Name` in a context of pod being deleted?

Reply:
https://godoc.org/k8s.io/api/admission/v1beta1#AdmissionRequest

It is the name of the pod. I have refactored the previous approach that was also taking `req.Name`, I just made the list of secret names more generic.